frontend/packages/topology/src/data-transforms/TopologyDataRetriever.tsx

```diff
@@ -18,34 +26,21 @@ const TopologyDataRetriever: React.FC<TopologyDataRetrieverProps> = ({ trafficDa
   const dataModelContext = React.useContext<ExtensibleModel>(ModelContext);
   const { namespace } = dataModelContext;
   const filters = useDisplayFilters();
+  const [resources, setResources] = React.useState<WatchK8sResults<TopologyResourcesObject>>();
+  const monitoringAlerts = useMonitoringAlerts(namespace);
   const showGroups = getFilterById(SHOW_GROUPS_FILTER_ID, filters)?.value ?? true;
   const resourcesList = React.useMemo<WatchK8sResources<any>>(
     () => (namespace && dataModelContext.extensionsLoaded ? dataModelContext.watchedResources : {}),
     [dataModelContext.extensionsLoaded, dataModelContext.watchedResources, namespace],
   );
-
-  const resources = useK8sWatchResources<TopologyResourcesObject>(resourcesList);
-  const [alertsResponse, alertsError, alertsLoading] = usePrometheusRulesPoll({ namespace });
-  const monitoringAlerts = React.useMemo(() => {
-    let alertData;
-    if (!alertsLoading && !alertsError) {
-      alertData = getAlertsAndRules(alertsResponse?.data).alerts;
-
-      // Don't update due to time changes
-      alertData.forEach((alert) => {
-        delete alert.activeAt;
-        if (alert.rule) {
-          delete alert.rule.evaluationTime;
-          delete alert.rule.lastEvaluation;
-          alert.rule.alerts &&
-            alert.rule.alerts.forEach((ruleAlert) => {
-              delete ruleAlert.activeAt;
-            });
-        }
-      });
+  const updateResources = _.debounce((newResources) => {
```

Comment:
Need to use the `debounce` hook otherwise this creates a new debounce function on ever update.